src/main/java/com/dataartisans/App.java

```diff
@@ -11,16 +14,64 @@
 import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
 
 public class App {
+    private static final String INPUT_PATH_PARAM = "inputPath";
+    private static final String OUTPUT_PATH_PARAM = "outputPath";
+
     public static void main( String[] args ) throws Exception {
-        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final ParameterTool params = ParameterTool.fromArgs(args);
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
 
-        DataStream<Event> eventStream = env.addSource(new EventGenerator());
+        DataStream<Event> eventStream = getStreamFromSource(env, params);
+
         eventStream.transform("OrderOperator", eventStream.getType(), new OrderOperator<>());
 
+        setStreamSink(eventStream, params);
+
         env.execute("Run Stream smoother");
     }
 
+    private static DataStream<Event> getStreamFromSource(StreamExecutionEnvironment env, ParameterTool appConfig) {
+        if(appConfig.has(INPUT_PATH_PARAM)) {
+            return env.readTextFile(appConfig.get(INPUT_PATH_PARAM)).map(new JsonToEventMapper());
```

Comment:
Open question: is `StreamExecutionEnvironment.readTextFile(path)` followed by a 'map to json' step appropriate to use here? Or should I prefer `StreamExecutionEnvironment.readFile(fileInputFormat, path)`?